src/main/java/com/example/online_shop_api/Service/UserService.java

```diff
@@ -218,4 +222,136 @@ private void saveThePurchasePriceInOrderProduct(List<OrderProduct> orderProducts
+        List<OrderProduct> currentOrderProducts = orderProductRepository.findAllByOrder(order);
+        Long currentOrderProductId;
+        try {
+            currentOrderProductId = orderProductRepository.findByOrderIdAndProductId(orderId, productId).getId();
+        } catch (RuntimeException e) {
+            throw new RuntimeException("Order/Product was not found!");
+        }
+        if (newQuantity == 0) {
+            orderProductRepository.deleteById(currentOrderProductId);
+            return;
+        }
+        for (OrderProduct op : currentOrderProducts) {
+            if (op.getProduct().getId().equals(productId)) {
+                op.setQuantity(newQuantity);
+                orderProductRepository.save(op);
+                return;
+            }
+        }
+    }
+
+    private boolean isQuantityAvailable(Product product, int newQuantity) {
+        return product.getQuantity() >= newQuantity;
+    }
+
+    public ResponseEntity<?> updateQuantity(Long productId, Long orderId, int quantity) {
+        try {
+            updateProductQuantity(orderId, productId, quantity);
+        } catch (RuntimeException e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        }
+        Order basketOrder = orderRepository.findById(orderId).get();
+        BigDecimal totalPrice = calculateTotalPrice(getBasketProducts(basketOrder));
+        List<ProductResponseDto> basketProducts = getBasketProducts(basketOrder);
+        return ResponseEntity.ok(new BasketResponse(basketProducts, totalPrice));
+    }
+
+    private String formatDeliveryAddress(User user) {
+        return user.getAddress().getCity().getName() + ", " + user.getAddress().getStreetName();
+    }
+
+    private BigDecimal calculateTotalPriceAndFillProductResponses(List<OrderProduct> products, List<ProductResponse> productResponses) {
+        BigDecimal total = BigDecimal.ZERO;
+        for (OrderProduct op : products) {
+            ProductResponse productResponse = new ProductResponse(op.getProduct().getName(), op.getQuantity());
+            productResponses.add(productResponse);
+            total = total.add(op.getProductPriceWhenPurchased().multiply(BigDecimal.valueOf(op.getQuantity())));
+        }
+        return total;
+    }
+
+    private UserOrdersResponse createUserOrderResponse(User user, Order order) {
+        UserOrdersResponse response = new UserOrdersResponse();
+        response.setOrderId(order.getId());
+        response.setDeliveryAddress(formatDeliveryAddress(user));
+        response.setOrderDate(order.getOrderDateTime());
+        response.setStatus(order.getStatus().getName());
+
+        List<OrderProduct> products = orderProductRepository.findAllByOrder_Id(order.getId());
+        List<ProductResponse> productResponses = new ArrayList<>();
+        BigDecimal price = calculateTotalPriceAndFillProductResponses(products, productResponses);
+
+        response.setPrice(price);
+        response.setProducts(productResponses);
```

Comment:
This will be an empty ArrayList, that is set on row 302

Reply:
I made a separate method for getting the products and their quantities.